static/app/views/alerts/rules/metric/ruleForm.tsx

```diff
@@ -454,6 +459,21 @@ class RuleFormContainer extends DeprecatedAsyncComponent<Props, State> {
     changedTriggerIndex?: number
   ) {
     const {comparisonType} = this.state;
+    // If we have an anomaly detection alert, then we don't need to validate the thresholds, but we do need to set them to 0
+    if (comparisonType === AlertRuleComparisonType.DYNAMIC) {
+      // NOTE: we don't support warning triggers for anomaly detection alerts yet
+      // once we do, uncomment this code and delete the next five uncommented lines:
+      // triggers.forEach(trigger => {
+      //   trigger.alertThreshold = 0;
+      // });
+      const criticalTriggerIndex = triggers.findIndex(
+        ({label}) => label === AlertRuleTriggerType.CRITICAL
+      );
+      const criticalTrigger = triggers[criticalTriggerIndex];
+      criticalTrigger.alertThreshold = 0;
+      const triggerErrors = new Map();
```

Comment:
nit: i'd move this to the top of the function and then you don't need to have it defined on 477 and 474, and can just return `triggerErrors`. I think that will also help the code read a little more cleanly as an "this is an overrde for the Dynamic alert types"